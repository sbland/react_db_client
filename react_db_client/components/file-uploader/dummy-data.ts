export const asyncFileUpload = async () => {};
export const onUpload = () => {};
