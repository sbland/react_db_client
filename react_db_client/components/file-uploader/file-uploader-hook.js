/* A react hook async request */

import { useState, useEffect } from 'react';

const ENV = process.env.NODE_ENV;

export const useFileUploader = ({ collectionId, documentId, fileType, onUpload, asyncUpload }) => {
  const [filesToUpload, setFilesToUpload] = useState(null);
  // const [selected, setSelected] = useState(null);
  // const [uploadSelected, setUploadSelected] = useState(false);
  const [uploadProgress, setUploadProgress] = useState(0);
  const [uploading, setUploading] = useState(false);
  const [error, setError] = useState(null);
  const [uploadComplete, setUploadComplete] = useState(false);
  // const [message, setMessage] = useState('');

  // handle upload selected
  useEffect(() => {
    if (filesToUpload && !uploading) {
      setUploading(true);
      setUploadProgress(filesToUpload.length);

      const promises = filesToUpload.map(async (file) => {
        await asyncUpload(file.data, collectionId, documentId, fileType, () => {});
        setFilesToUpload((prev) => prev.filter((f) => f.name !== file.name));
        setUploadProgress((prev) => prev - 1);
      });
      Promise.all(promises)
        .then((responses) => {
          setUploadComplete(true);
          onUpload(responses);
        })
        .catch((e) => {
          if (ENV === 'development') console.log(e);
          setError('Upload Failed');
        })
        .finally(() => {
          setUploading(false);
          setFilesToUpload(null);
        });
    }
    return () => {
      // TODO: handle cancel upload
    };
  }, [filesToUpload, uploading, collectionId, documentId, fileType, onUpload, uploadProgress]);

  const uploadFiles = (fileList) => {
    setFilesToUpload(fileList);
  };

  return {
    uploadFiles,
    uploading,
    uploadProgress,
    uploadComplete,
    error,
  };
};

