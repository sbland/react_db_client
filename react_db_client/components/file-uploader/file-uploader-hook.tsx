/* A react hook async request */
import { useState, useEffect, useMemo } from 'react';
import { EFileType, IFile } from '@react_db_client/constants.client-types';

const ENV = process.env.NODE_ENV;

export interface IUseFileUploaderArgs {
  fileType: EFileType;
  onUpload: (responses: unknown) => void;
  asyncFileUpload: (
    data: File,
    fileType: EFileType,
    callback: () => void,
    metaData: Partial<IFile>
  ) => Promise<void>;
}

export interface IUseFileUploaderReturn {
  uploadFiles: (filelist: IFile[]) => void;
  uploading: boolean;
  uploadProgress: number;
  uploadComplete: boolean;
  error: string;
  selectedFiles: IFile[];
  handleFilesSelectedForUpload: (
    newFiles: FileList | null,
    options?: {
      instantUpload?: boolean;
    }
  ) => void;
  totalFilesToUpload: null | number;
  fileSelectionComplete: boolean;
}

const promiseAllInSeries = async (iterable) => {
  let output: Array<any> = [];
  for (const x of iterable) {
    await x().then((response) => output.push(response));
  }
  return output;
};

export const useFileUploader = ({
  fileType,
  onUpload,
  asyncFileUpload,
}: IUseFileUploaderArgs): IUseFileUploaderReturn => {
  const [selectedFiles, setSelectedFiles] = useState<IFile[]>([]);
  const [filesToUpload, setFilesToUpload] = useState<null | IFile[]>(null);
  const [uploadProgress, setUploadProgress] = useState(0);
  const [uploading, setUploading] = useState(false);
  const [error, setError] = useState<string | null>(null);
  const [uploadComplete, setUploadComplete] = useState(false);
  const [totalFilesToUpload, setTotalFilesToUpload] = useState<number | null>(
    null
  );
  const [filesMetaData, setFilesMetaData] = useState<Partial<IFile>[]>([]);

  // Needed to disable the upload button until file selection complete
  const fileSelectionComplete = useMemo(
    () =>
      filesMetaData.length > 0 && filesMetaData.length === selectedFiles.length,
    [filesMetaData, selectedFiles]
  );

  // handle upload selected
  useEffect(() => {
    if (filesToUpload && !uploading && !error && fileSelectionComplete) {
      setUploading(true);
      setUploadProgress(filesToUpload.length);
      const promises = filesToUpload.map((file, i) => async () => {
        if (!file.data) throw Error('Missing file data');
        const metaData: Partial<IFile> = filesMetaData[i];
        await asyncFileUpload(file.data, fileType, () => {}, metaData);
        setFilesToUpload(
          (prev) => prev?.filter((f) => f.name !== file.name) || null
        );
        setUploadProgress((prev) => prev - 1);
        return `Uploaded ${file.name}`;
      });
      promiseAllInSeries(promises)
        .then((responses) => {
          setUploadComplete(true);
          setUploading(false);
          setFilesToUpload(null);
          setSelectedFiles([]);
          onUpload(responses);
          setTotalFilesToUpload(filesToUpload.length);
          setFilesMetaData([]);
        })
        .catch((e) => {
          setUploadComplete(true);
          setUploading(false);
          if (ENV === 'development') console.log(e);
          setError('Upload Failed');
        });
    }
    return () => {
      // TODO: handle cancel upload
    };
  }, [
    filesToUpload,
    uploading,
    fileType,
    onUpload,
    uploadProgress,
    fileSelectionComplete,
  ]);

  const uploadFiles = (fileList: IFile[]) => {
    setError(null);
    setTotalFilesToUpload(fileList.length);
    setFilesToUpload(fileList);
  };

  const handleFilesSelectedForUpload = (
    newFiles: FileList | null,
    options: { instantUpload?: boolean } = {}
  ) => {
    const files = [...(newFiles || [])];
    const newSelectedFiles: IFile[] = files.map((f: File) => {
      return {
        uid: f.name,
        name: f.name,
        label: f.name,
        data: f,
        filePath: '',
        fileType,
      };
    });
    setSelectedFiles(newSelectedFiles);

    files.forEach((f) => {
      const reader = new FileReader();
      if (fileType === EFileType.IMAGE) {
        reader.onload = () => {
          const data = reader.result as string;
          const img = new Image();
          img.onload = () => {
            setFilesMetaData((prev) => [
              ...prev,
              { width: img.width, height: img.height, name: f.name },
            ]);
          };
          img.src = data;
        };
        reader.readAsDataURL(f);
      } else {
        setFilesMetaData((prev) => [...prev, { name: f.name }]);
      }
    });
    if (options.instantUpload) uploadFiles(newSelectedFiles);
  };

  return {
    uploadFiles,
    uploading,
    uploadProgress,
    uploadComplete,
    error,
    selectedFiles,
    handleFilesSelectedForUpload,
    totalFilesToUpload,
    fileSelectionComplete,
  } as IUseFileUploaderReturn;
};
