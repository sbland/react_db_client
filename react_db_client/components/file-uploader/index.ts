export { FileUploader } from './file-uploader';
export { FileUploaderSimple } from './file-uploader-simple';

export type { IFileUploaderProps } from './file-uploader-simple';
