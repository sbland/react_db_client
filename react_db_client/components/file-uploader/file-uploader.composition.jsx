import React from 'react';
import { CompositionWrapDefault } from '@samnbuk/react_db_client.helpers.composition-wraps';
import { FileUploader } from './file-uploader';

const asyncUpload = async () => {}
const onUpload = () => {};

const defaultProps = {
  collectionId: 'demoCollectionId',
  documentId: 'demoDocumentId',
  fileType: 'image',
  asyncUpload,
  onUpload,
};


export const BasicFileUploader  = () => (
  <CompositionWrapDefault height="4rem" width="8rem">
    <FileUploader
      {...defaultProps}
      required
    />
  </CompositionWrapDefault>
);
