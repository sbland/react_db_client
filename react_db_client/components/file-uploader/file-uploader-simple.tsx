import PropTypes from 'prop-types';
import React, { useState, useMemo, useEffect, useRef } from 'react';
import { EFileType, IFile } from '@react_db_client/constants.client-types';
import { useFileUploader } from './file-uploader-hook';
import { FileUploadBtn, FileUploaderButtonsWrap } from './styles';
// import UploadIcon from './assets/UploadIcon.svg'; // TODO: implement upload icon

const fileTypesToInputAccept = (fileType: EFileType) => {
  switch (fileType) {
    case EFileType.IMAGE:
      return 'image/*';
    case EFileType.DOCUMENT:
      return '.pdf,.doc,.docx,.txt,.odt';
    case EFileType.DATA:
      return '.xml,.csv,.xlsx,.json,.yaml';
    default:
      return fileType || '*';
  }
};

export interface IFileUploaderProps {
  fileType: EFileType;
  onUpload: (responses: unknown) => void;
  asyncFileUpload: (data: File, fileType: EFileType, callback: () => void) => Promise<void>;
}

export const FileUploaderSimple = ({
  fileType: fileTypeIn,
  onUpload,
  asyncFileUpload,
}: IFileUploaderProps) => {
  const [selectedFiles, setSelectedFiles] = useState<IFile[]>([]);
  const [fileType, setFileType] = useState(fileTypeIn || EFileType.IMAGE);
  const inputRef = useRef<HTMLInputElement>(null);

  const { uploadFiles, uploading, uploadProgress, uploadComplete, error } = useFileUploader({
    fileType,
    asyncFileUpload,
    onUpload,
  });

  const message = useMemo(() => {
    if (error) return error;
    if (uploading) return `Uploading ${uploadProgress} of ${selectedFiles.length}`;
    if (uploadComplete) return 'Upload complete';
    return '';
  }, [error, uploading, uploadComplete, uploadProgress, selectedFiles]);

  const handleFilesSelected = (e) => {
    const newSelectedFiles: IFile[] = [...e.target.files].map((f: File) => ({
      uid: f.name,
      name: f.name,
      label: f.name,
      data: f,
      filePath: '',
      fileType,
    }));
    setSelectedFiles(newSelectedFiles);
  };

  useEffect(() => {
    if (selectedFiles && selectedFiles[0]) {
      uploadFiles(selectedFiles);
    }
  }, [selectedFiles]);

  return (
    <div className="fileUploader">
      <FileUploaderButtonsWrap>
        <FileUploadBtn onClick={(e) => inputRef.current?.click()}>
          <label
            className={`button ${
              (selectedFiles && selectedFiles.length > 0) || uploading ? 'button-one' : 'button-two'
            } fileUploader_selectBtn`}
          >
            {/* <img src={UploadIcon} alt="Upload" /> */}
            Upload
            <input
              // Hide html input so we can override style
              ref={inputRef}
              style={{ display: 'none' }}
              name="file"
              id="file"
              className="fileUploader_fileInput"
              type="file"
              accept={fileTypesToInputAccept(fileType)}
              // multiple={multiple}
              onChange={handleFilesSelected}
              // onInput={console.log}
              // value={selectedFilesRaw}
              // disabled={uploading}
              multiple
            />
          </label>
        </FileUploadBtn>
        {message}
      </FileUploaderButtonsWrap>
      {/* If file type not defined provide option to user */}
    </div>
  );
};

FileUploaderSimple.propTypes = {
  fileType: PropTypes.oneOf(['image', 'document', 'data', '*']),
  onUpload: PropTypes.func.isRequired,
  asyncFileUpload: PropTypes.func.isRequired,
};

FileUploaderSimple.defaultProps = {
  fileType: '*',
};
