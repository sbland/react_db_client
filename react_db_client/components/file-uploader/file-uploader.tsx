import PropTypes from 'prop-types';
import React, { useState, useMemo } from 'react';
import {
  IHeading,
  StyledSelectList,
} from '@react_db_client/components.styled-select-list';
import {
  EFileType,
  EFilterType,
  IFile,
} from '@react_db_client/constants.client-types';

import { useFileUploader } from './file-uploader-hook';
import { FileUploaderButtonsWrap, FileUploaderSelectButton } from './styles';

const fileTypesToInputAccept = (fileType: EFileType) => {
  switch (fileType) {
    case EFileType.IMAGE:
      return 'image/*';
    case EFileType.DOCUMENT:
      return '.pdf,.doc,.docx,.txt,.odt';
    case EFileType.DATA:
      return '.xml,.csv,.xlsx,.json,.yaml';
    default:
      return fileType || '*';
  }
};

const fileListHeadings: IHeading[] = [
  { uid: 'name', label: 'Name', type: EFilterType.text },
];

export interface IFileUploaderProps {
  fileType: EFileType;
  onUpload: (responses: unknown) => void;
  asyncFileUpload: (
    data: File,
    fileType: EFileType,
    callback: () => void,
    metaData: Partial<IFile>
  ) => Promise<void>;
}

export const FileUploader = ({
  fileType: fileTypeIn,
  onUpload,
  asyncFileUpload,
}: IFileUploaderProps) => {
  const [fileType, setFileType] = useState(fileTypeIn || EFileType.IMAGE);

  const {
    selectedFiles,
    handleFilesSelectedForUpload,
    uploadFiles,
    uploading,
    uploadProgress,
    uploadComplete,
    totalFilesToUpload,
    error,
    fileSelectionComplete,
  } = useFileUploader({
    fileType,
    asyncFileUpload,
    onUpload,
  });

  const message = useMemo(() => {
    if (error) return error;
    if (uploading)
      return `Uploading ${uploadProgress} of ${totalFilesToUpload}`;
    if (uploadComplete) return 'Upload complete';
    return '';
  }, [error, uploading, uploadComplete, uploadProgress, totalFilesToUpload]);

  return (
    <div className="fileUploader">
      {message}
      <FileUploaderButtonsWrap className="fileUploader_btnsWrap">
        <label
          className={`button ${
            totalFilesToUpload || uploading ? 'button-one' : 'button-two'
          } fileUploader_selectBtn`}
        >
          Select Files
          <input
            // Hide html input so we can override style
            style={{ display: 'none' }}
            name="file"
            id="file"
            className="fileUploader_fileInput"
            type="file"
            accept={fileTypesToInputAccept(fileType)}
            onChange={(e) => handleFilesSelectedForUpload(e.target.files)}
            multiple
          />
        </label>

        <FileUploaderSelectButton
          type="button"
          disabled={
            !fileSelectionComplete ||
            !selectedFiles ||
            selectedFiles.length === 0 ||
            uploading
          }
          className="button-two uploadBtn"
          onClick={() => uploadFiles(selectedFiles)}
        >
          Upload
        </FileUploaderSelectButton>
      </FileUploaderButtonsWrap>
      {/* If file type not defined provide option to user */}
      {!fileTypeIn && (
        <select
          onChange={(e) => setFileType(e.target.value as EFileType)}
          value={fileType}
        >
          <option value={EFileType.IMAGE}>Image</option>
          <option value={EFileType.DOCUMENT}>Document</option>
        </select>
      )}
      <p>Files ready to upload:</p>
      {fileType === EFileType.IMAGE && (
        <div className="fileUploader_imagePreview">
          {selectedFiles &&
            selectedFiles.map((file) => (
              <img
                key={file.name}
                width="100px"
                src={URL.createObjectURL(file.data as File)}
                alt={file.name}
              />
            ))}
        </div>
      )}
      <StyledSelectList
        listInput={selectedFiles || []}
        headings={fileListHeadings}
        // TODO: Handle remove from selection
        handleSelect={(uid) => {}}
        currentSelection={undefined}
        // limitHeight
        selectionField="uid"
        // autoWidth
      />
    </div>
  );
};

FileUploader.propTypes = {
  fileType: PropTypes.oneOf(['image', 'document', 'data', '*']),
  onUpload: PropTypes.func.isRequired,
  asyncFileUpload: PropTypes.func.isRequired,
};

FileUploader.defaultProps = {
  fileType: '*',
};
