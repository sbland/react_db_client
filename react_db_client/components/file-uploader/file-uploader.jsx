import PropTypes from 'prop-types';
import React, { useState, useMemo } from 'react';
import { StyledSelectList } from '@samnbuk/react_db_client.components.styled-select-list';
import { filterTypes } from '@samnbuk/react_db_client.constants.client-types';
// import ItemList from '../ItemList/ItemList';

import { useFileUploader } from './file-uploader-hook';
import './style.scss';

const fileTypesToInputAccept = (fileType) => {
  switch (fileType) {
    case 'image':
      return 'image/*';
    case 'document':
      return '.pdf,.doc,.docx,.txt,.odt';
    case 'data':
      return '.xml,.csv,.xlsx,.json,.yaml';
    default:
      return fileType || '*';
  }
};

const fileListHeadings = [{ uid: 'name', label: 'Name', type: filterTypes.text }];

export const FileUploader = ({
  collectionId,
  documentId,
  fileType: fileTypeIn,
  onUpload,
  asyncUpload,
}) => {
  const [selectedFiles, setSelectedFiles] = useState(null);
  const [fileType, setFileType] = useState(fileTypeIn || 'image');

  const { uploadFiles, uploading, uploadProgress, uploadComplete, error } = useFileUploader({
    collectionId,
    documentId,
    fileType,
    asyncUpload,
    onUpload,
  });

  const message = useMemo(() => {
    if (error) return error;
    if (uploading) return `Uploading ${uploadProgress} of ${selectedFiles.length}`;
    if (uploadComplete) return 'Upload complete';
    return '';
  }, [error, uploading, uploadComplete, uploadProgress, selectedFiles]);

  const handleFilesSelected = (e) => {
    const newSelectedFiles = [...e.target.files].map((f) => ({ uid: f.uid, name: f.name, data: f }));
    setSelectedFiles(newSelectedFiles);
  };

  return (
    <div className="fileUploader">
      {message}
      <div className="fileUploader_btnsWrap">
        <label
          className={`button ${
            (selectedFiles && selectedFiles.length > 0) || uploading ? 'button-one' : 'button-two'
          } fileUploader_selectBtn`}
        >
          Select Files
          <input
            // Hide html input so we can override style
            style={{ display: 'none' }}
            name="file"
            id="file"
            className="fileUploader_fileInput"
            type="file"
            accept={fileTypesToInputAccept(fileType)}
            // multiple={multiple}
            onChange={handleFilesSelected}
            // onInput={console.log}
            // value={selectedFilesRaw}
            // disabled={uploading}
            multiple
          />
        </label>

        <button
          type="button"
          disabled={!selectedFiles || selectedFiles.length === 0 || uploading}
          className="button-two uploadBtn"
          // className={(selected) ? 'button-two' : 'button-one'}
          onClick={() => uploadFiles(selectedFiles)}
        >
          Upload
        </button>
      </div>
      {/* If file type not defined provide option to user */}
      {!fileTypeIn && (
        <select onChange={(e) => setFileType(e.target.value)} value={fileType}>
          <option value="image">Image</option>
          <option value="document">Document</option>
        </select>
      )}
      <StyledSelectList
        listInput={selectedFiles || []}
        headings={fileListHeadings}
        handleSelect={(uid) => {}}
        currentSelection={null}
        // limitHeight
        selectionField="uid"
        autoWidth
      />
      {/* <ItemList
        items={
          !selectedFiles
            ? []
            : selectedFiles.map((s) => ({
                uid: s.name,
                name: s.name,
                type: 'button',
              }))
        }
      /> */}
    </div>
  );
};

FileUploader.propTypes = {
  collectionId: PropTypes.string.isRequired,
  documentId: PropTypes.string.isRequired,
  fileType: PropTypes.oneOf(['image', 'document', 'data', '*']),
  onUpload: PropTypes.func.isRequired,
  asyncUpload: PropTypes.func.isRequired,
};

FileUploader.defaultProps = {
  fileType: '*',
};
