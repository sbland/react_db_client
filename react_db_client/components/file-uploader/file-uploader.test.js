import '@samnbuk/react_db_client.testing.enzyme-setup';
import React from 'react';
import { mount, shallow } from 'enzyme';

import { FileUploader } from './file-uploader';
import * as compositions from './file-uploader.composition';
import { StyledSelectList } from '@samnbuk/react_db_client.components.styled-select-list';

const asyncUpload = jest.fn();
const onUpload = jest.fn();

const defaultProps = {
  collectionId: 'demoCollectionId',
  documentId: 'demoDocumentId',
  fileType: 'image',
  asyncUpload,
  onUpload,
};

describe('file-uploader', () => {
  beforeEach(() => {
    asyncUpload.mockClear();
    onUpload.mockClear();
  });
  test('Renders', () => {
    shallow(<FileUploader {...defaultProps} />);
  });

  describe('Compositions', () => {
    Object.entries(compositions).forEach(([name, Composition]) => {
      test(name, () => {
        mount(<Composition />);
      });
    });
  });
  describe('shallow renders', () => {
    test('Matches Snapshot', () => {
      const component = shallow(<FileUploader {...defaultProps} />);
      const tree = component.debug();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Unit Testing', () => {
    let component;
    beforeEach(() => {
      component = mount(<FileUploader {...defaultProps} />);
    });
    test('should add files to selected on file select', () => {
      const fileInput = component.find('.fileUploader_fileInput');
      expect(fileInput).toBeTruthy();
      const event = {
        target: {
          name: 'pollName',
          files: [
            { uid: 'fileA', name: 'fileA' },
            { uid: 'fileB', name: 'fileB' },
          ],
        },
      };
      fileInput.simulate('change', event);
      const fileList = component.find(StyledSelectList);
      expect(fileList.props().listInput.length).toEqual(2);
      const tree = component.debug();
      expect(tree).toMatchSnapshot();
    });

    // test('should send a xhr request on clicking the upload button', () => {
    //   // TODO: Test we call uploader hook here
    //   const uploadBtn = component.find('.uploadBtn');
    //   uploadBtn.simulate('click');
    // });
  });
});
