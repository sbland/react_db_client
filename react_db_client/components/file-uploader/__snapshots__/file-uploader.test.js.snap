// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file-uploader Unit Testing should add files to selected on file select 1`] = `
"<FileUploader collectionId=\\"demoCollectionId\\" documentId=\\"demoDocumentId\\" fileType=\\"image\\" asyncUpload={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} onUpload={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }}>
  <div className=\\"fileUploader\\">
    <div className=\\"fileUploader_btnsWrap\\">
      <label className=\\"button button-one fileUploader_selectBtn\\">
        Select Files
        <input style={{...}} name=\\"file\\" id=\\"file\\" className=\\"fileUploader_fileInput\\" type=\\"file\\" accept=\\"image/*\\" onChange={[Function: handleFilesSelected]} multiple={true} />
      </label>
      <button type=\\"button\\" disabled={false} className=\\"button-two uploadBtn\\" onClick={[Function: onClick]}>
        Upload
      </button>
    </div>
    <StyledSelectList listInput={{...}} headings={{...}} handleSelect={[Function: handleSelect]} currentSelection={{...}} selectionField=\\"uid\\" autoWidth={true} limitHeight={0} customParsers={{...}}>
      <div className=\\"styledList\\" style={{...}}>
        <div style={{...}} className=\\"styledList_headings\\">
          <div className=\\"styledList_heading\\" style={{...}}>
            Name
          </div>
        </div>
        <div style={{...}} className=\\"styledList_items \\" role=\\"list\\">
          <ListItem data={{...}} currentSelection={{...}} selectionField=\\"uid\\" handleSelect={[Function: handleSelect]} headings={{...}} columnWidths={{...}} customParsers={{...}} isSelected={false} classes={{...}}>
            <button className=\\"styledList_itemBtn styledList_itemBtn \\" type=\\"button\\" onClick={[Function: onClick]} role=\\"listitem\\">
              <div style={{...}} className=\\"styledList_itemCell \\">
                fileA
              </div>
            </button>
          </ListItem>
          <ListItem data={{...}} currentSelection={{...}} selectionField=\\"uid\\" handleSelect={[Function: handleSelect]} headings={{...}} columnWidths={{...}} customParsers={{...}} isSelected={false} classes={{...}}>
            <button className=\\"styledList_itemBtn styledList_itemBtn \\" type=\\"button\\" onClick={[Function: onClick]} role=\\"listitem\\">
              <div style={{...}} className=\\"styledList_itemCell \\">
                fileB
              </div>
            </button>
          </ListItem>
        </div>
      </div>
    </StyledSelectList>
  </div>
</FileUploader>"
`;

exports[`file-uploader shallow renders Matches Snapshot 1`] = `
"<div className=\\"fileUploader\\">
  <div className=\\"fileUploader_btnsWrap\\">
    <label className=\\"button button-two fileUploader_selectBtn\\">
      Select Files
      <input style={{...}} name=\\"file\\" id=\\"file\\" className=\\"fileUploader_fileInput\\" type=\\"file\\" accept=\\"image/*\\" onChange={[Function: handleFilesSelected]} multiple={true} />
    </label>
    <button type=\\"button\\" disabled={true} className=\\"button-two uploadBtn\\" onClick={[Function: onClick]}>
      Upload
    </button>
  </div>
  <StyledSelectList listInput={{...}} headings={{...}} handleSelect={[Function: handleSelect]} currentSelection={{...}} selectionField=\\"uid\\" autoWidth={true} limitHeight={0} customParsers={{...}} />
</div>"
`;
